rcl_yaml_param_parser/test/test_parse.cpp

```diff
@@ -0,0 +1,417 @@
+
+  // Check bad sequence type for int
+  seq_data_type = DATA_TYPE_STRING;
+  EXPECT_EQ(
+    RCUTILS_RET_ERROR,
+    parse_value(event, is_seq, node_idx, parameter_idx, &seq_data_type, params_st)) <<
+    rcutils_get_error_string().str;
+  EXPECT_EQ(nullptr, params_st->params[0].parameter_values[0].integer_array_value);
+
+  // Check proper sequence type
+  seq_data_type = DATA_TYPE_UNKNOWN;
+  EXPECT_EQ(
+    RCUTILS_RET_OK,
+    parse_value(event, is_seq, node_idx, parameter_idx, &seq_data_type, params_st)) <<
+    rcutils_get_error_string().str;
+  ASSERT_NE(nullptr, params_st->params[0].parameter_values[0].integer_array_value);
+  EXPECT_EQ(42, params_st->params[0].parameter_values[0].integer_array_value->values[0]);
+  allocator.deallocate(
+    params_st->params[0].parameter_values[0].integer_array_value->values, allocator.state);
+  allocator.deallocate(
+    params_st->params[0].parameter_values[0].integer_value, allocator.state);
+  params_st->params[0].parameter_values[0].integer_array_value = nullptr;
+
+  // double value
+  yaml_char_t double_value[] = "3.14159";
+  const size_t double_value_length = sizeof(double_value) / sizeof(double_value[0]);
+  event.data.scalar.value = double_value;
+  event.data.scalar.length = double_value_length;
+
+  // Check bad sequence type for double
+  seq_data_type = DATA_TYPE_STRING;
+  EXPECT_EQ(
+    RCUTILS_RET_ERROR,
+    parse_value(event, is_seq, node_idx, parameter_idx, &seq_data_type, params_st)) <<
+    rcutils_get_error_string().str;
+  EXPECT_EQ(nullptr, params_st->params[0].parameter_values[0].integer_array_value);
+
+  // Check proper sequence type
+  seq_data_type = DATA_TYPE_UNKNOWN;
+  EXPECT_EQ(
+    RCUTILS_RET_OK,
+    parse_value(event, is_seq, node_idx, parameter_idx, &seq_data_type, params_st)) <<
+    rcutils_get_error_string().str;
+  ASSERT_NE(nullptr, params_st->params[0].parameter_values[0].double_array_value);
+  EXPECT_EQ(3.14159, params_st->params[0].parameter_values[0].double_array_value->values[0]);
+  allocator.deallocate(
+    params_st->params[0].parameter_values[0].double_array_value->values, allocator.state);
+  allocator.deallocate(
+    params_st->params[0].parameter_values[0].double_array_value, allocator.state);
+  params_st->params[0].parameter_values[0].double_array_value = nullptr;
+
+  // double value
+  yaml_char_t string_value[] = "hello, I am a string";
+  const size_t string_value_length = sizeof(string_value) / sizeof(string_value[0]);
+  event.data.scalar.value = string_value;
+  event.data.scalar.length = string_value_length;
+
+  // Check bad sequence type for string
+  seq_data_type = DATA_TYPE_BOOL;
+  EXPECT_EQ(
+    RCUTILS_RET_ERROR,
+    parse_value(event, is_seq, node_idx, parameter_idx, &seq_data_type, params_st)) <<
+    rcutils_get_error_string().str;
+  EXPECT_EQ(nullptr, params_st->params[0].parameter_values[0].integer_array_value);
+
+  // Check proper sequence type
+  seq_data_type = DATA_TYPE_UNKNOWN;
```

Comment:
If `DATA_TYPE_UNKNOWN` makes sure any data type can be used, IMHO in the tests it should be added the correct data type for each test, and at the end one test that outlines that this type can be used for any of the types